FacesofLondon/app/src/main/java/com/novoda/wallpaper/droidcon/TimeOfDayAssetRefresher.java

```diff
@@ -0,0 +1,27 @@
+package com.novoda.wallpaper.droidcon;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+class TimeOfDayAssetRefresher extends BroadcastReceiver {
+
+    private final TimeOfDayCalculator calculator;
+    private final LondonParallaxWallpaperRenderer renderer;
+    private TimeOfDay currentTime;
+
+    TimeOfDayAssetRefresher(LondonParallaxWallpaperRenderer renderer, TimeOfDayCalculator calculator) {
+        this.renderer = renderer;
+        this.calculator = calculator;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        TimeOfDay timeOfDay = calculator.currentTimeOfDay();
+        if (currentTime != timeOfDay) {
```

Comment:
Come and think of it, the first time onReceive() gets called `currentTime` is always going to be `null`. I must have forgotten to initialise it. 

I think there is no reason to be worried (famous last words)